EJ2-Documentation(846855) - UG documentation for DetailTemplate

Gayathri4135 opened the pull request:
Modified the sample and content of Rendering custom component, Expand by external button, Limitation of detail Template and added a new topic Customize the detail template icon.

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [Ej2-846855-DetailTemplate](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/315/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [Ej2-846855-DetailTemplate](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/315/)<br/>**

BalajiSGrid commented:
Technical has been completed

Gayathri4135 commented:
Changes reviewed in the PR: https://github.com/syncfusion-content/ej2-grid-docs/pull/38
The reviewed changes :

1. Need to add space in "the[detailTemplate]"
Add the extra spaces.
2. Replace the platform name into "##Platform_Name## "
Replaced the platform name into ##Platform_Name## .
3. Grid data is empty on detailTemplate
In the sample,grid data is empty on detailTemplate It can be resolved by using return function.
4. Remove the unnecessary datasource.
Removed the unnecessary datasource.
5. use"e" instead of "DetailDataBoundEventArgs" in detailDataBound event's argument.
    Changed "e" instead of "DetailDataBoundEventArgs" in detailDataBound event's argument.
    
    hotfix version 23.1.36 PR: https://github.com/syncfusion-content/ej2-grid-docs/pull/74

MaithiliyK reviewed (approved):
changes are fine

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [Ej2-846855-DetailTemplate](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/352/)<br/>**
